framework/db/BaseActiveRecord.php

```diff
@@ -236,10 +236,9 @@ public function optimisticLock()
      */
     public function __get($name)
     {
-        if (isset($this->_attributes[$name]) || array_key_exists($name, $this->_attributes)) {
-            return $this->_attributes[$name];
-        } elseif ($this->hasAttribute($name)) {
-            return null;
+        // getAttribute has better performance than hasAttribute for attributes that are set.
+        if ((null !== $value = $this->getAttribute($name)) || $this->hasAttribute($name)) {
```

Comment:
Isn't that slowing down everything by duration of `getAttribute()` when it's not set?